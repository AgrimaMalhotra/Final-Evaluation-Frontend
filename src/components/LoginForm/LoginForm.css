.login-page {
  display: flex;
  height: 100vh;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.image-section {
  display: flex;
  flex-direction: column;
  width: 60vw;
  padding-top: 10vh;
  background: rgb(236, 234, 250);
}

.image-section p {
  margin-left: 10vw;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  font-size: 2.5rem;
  font-weight: 700;
  color: rgb(67, 67, 67);
  margin-bottom: 15px;
}

.image-section img {
  width: 50%;
  align-self: center;
}

.login-section {
  width: 40vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  background: rgb(42, 42, 42);
}

.login-section>p {
  font-size: 2rem;
  font-weight: 600;
  margin-top: 15vh;
  margin-bottom: 15vh;
}

.login-section p {
  color: white;
}

.login-form {
  font-weight: 300;
  font-size: 0.9rem;
}

.login-form input {
  width: 16vw;
  height: 4vh;
  border-radius: 6px;
  border: none;
  padding-left: 10px;
  padding-right: 10px;
  outline: none;
  margin-bottom: 14px;
  margin-top: 8px;
}

.login-form button {
  width: 16vw;
  height: 4vh;
  border-radius: 6px;
  border: none;
  margin-top: 22px;
  margin-bottom: 15px;
  background: rgb(135, 36, 168);
  background: linear-gradient(90deg, rgba(135, 36, 168, 1) 0%, rgb(45, 27, 104) 100%);
  color: white;
  font-family: Montserrat;
  font-weight: 500;
  cursor: pointer;
}

.forgot-pwd {
  text-align: center;
  text-decoration: underline;
  cursor: pointer;
}