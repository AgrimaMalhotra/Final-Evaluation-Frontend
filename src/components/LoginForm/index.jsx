import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { makeAuthRequest } from '../../utils/makeRequest/makeRequest';
import { LOGIN_USER } from '../../constants/apiEndPoints';
import './LoginForm.css';
import { DASHBOARD_ROUTE } from '../../constants/routes';
import CmsImage from '../CmsImage';

const LoginForm = () => {
  const navigate = useNavigate();
  const [userDetails, setUserDetails] = useState({ email: '', password: ''});
  const [, setError] = useState(null);
  
  const handleInputChange = (event) => {
    const { name, value } = event.target;
    setUserDetails(prevState => ({ ...prevState, [name]: value }));
  };
  const handleSubmit = async () => {
    try {
      const responseData = await makeAuthRequest(LOGIN_USER(),navigate, {
        data: {
          ...userDetails,
        },
      });
      localStorage.setItem('auth_token', responseData.token);
      navigate(DASHBOARD_ROUTE);
     
    } catch (error) {
      console.log(error);
      setError(error);
      setTimeout(() => {
        setError(null);
      }, 4000);
    }
  };
  return (
    <div className='login-page'>
      <CmsImage />
      <div className='login-section'>
        <p>Login to your CMS+ account</p>
        <div className='login-form'>
          <p>Email</p>
          <input type="email"
            name="email"
            value={userDetails.email}
            onChange={handleInputChange} />

          <p>Password</p>
          <input type="password"
            name="password"
            value={userDetails.password}
            onChange={handleInputChange} />

          <br />

          <button
            type="submit"
            disabled={ userDetails.password === ''|| userDetails.email=== ''}
            onClick={handleSubmit}>Login</button>
          <p className='forgot-pwd'>Forgot password?</p>
        </div>
      </div>
    </div>
  );
};

export default LoginForm;