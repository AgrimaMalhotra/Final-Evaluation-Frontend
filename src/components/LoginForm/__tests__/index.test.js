import React from 'react';
import {render} from '@testing-library/react';
import LoginForm from '../index';
jest.mock('react-router-dom', () => ({ useNavigate: () => jest.fn() }));
describe('LoginForm', () => {
  it('should render correctly', () => {
    const component = render(<LoginForm />);
    expect(component).toMatchSnapshot();
  });
});
