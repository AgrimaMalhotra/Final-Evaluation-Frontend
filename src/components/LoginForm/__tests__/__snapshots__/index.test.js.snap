// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-page"
      >
        <div
          class="logo-image"
        >
          <p>
            Design APIs fast,
          </p>
          <p>
            Manage content easily
          </p>
          <img
            src="undraw-upload-re-pasx.png"
          />
        </div>
        <div
          class="login-section"
        >
          <p>
            Login to your CMS+ account
          </p>
          <div
            class="login-form"
          >
            <p>
              Email
            </p>
            <input
              name="email"
              type="email"
              value=""
            />
            <p>
              Password
            </p>
            <input
              name="password"
              type="password"
              value=""
            />
            <br />
            <button
              disabled=""
              type="submit"
            >
              Login
            </button>
            <p
              class="forgot-pwd"
            >
              Forgot password?
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-page"
    >
      <div
        class="logo-image"
      >
        <p>
          Design APIs fast,
        </p>
        <p>
          Manage content easily
        </p>
        <img
          src="undraw-upload-re-pasx.png"
        />
      </div>
      <div
        class="login-section"
      >
        <p>
          Login to your CMS+ account
        </p>
        <div
          class="login-form"
        >
          <p>
            Email
          </p>
          <input
            name="email"
            type="email"
            value=""
          />
          <p>
            Password
          </p>
          <input
            name="password"
            type="password"
            value=""
          />
          <br />
          <button
            disabled=""
            type="submit"
          >
            Login
          </button>
          <p
            class="forgot-pwd"
          >
            Forgot password?
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
